### ISSUE TITLE
Callbacks documentation not showing bullet points correctly

### ISSUE THREAD

**Issue by xhlulu** [keras-team/keras#11673]
The current [documentation on callbacks](https://keras.io/callbacks/) isn't showing bullet points correctly under the "Arguments" section of a few models. Here's the example for `ModelCheckpoint`:

> filepath: string, path to save the model file. monitor: quantity to monitor. verbose: verbosity mode, 0 or 1. save_best_only: if save_best_only=True, the latest best model according to the quantity monitored will not be overwritten. mode: one of {auto, min, max}. If save_best_only=True, the decision to overwrite the current save file is made based on either the maximization or the minimization of the monitored quantity. For val_acc, this should be max, for val_loss this should be min, etc. In auto mode, the direction is automatically inferred from the name of the monitored quantity. save_weights_only: if True, then only the model's weights will be saved (model.save_weights(filepath)), else the full model is saved (model.save(filepath)). period: Interval (number of epochs) between checkpoints.

Looking at the source code, the docstring seems to be organized correctly:
https://github.com/keras-team/keras/blob/dc9e510192d0a8a6f6943cd46e9554364d4dcdd2/keras/callbacks.py#L371-L390

It is however showing up correctly for other models, e.g. `ProgbarLogger`:

> # Arguments
> * __count_mode:__ One of "steps" or "samples". Whether the progress bar should count samples seen or steps (batches) seen.
> * __stateful_metrics:__ Iterable of string names of metrics that should not be averaged over an epoch. Metrics in this list will be logged as-is. All others will be averaged over time (e.g. loss, etc).

**Comment by gabrieldemarmiesse** [comment]
Thank you for reporting it. Contributions are welcome. It's really easy to break the formatting of the documentation. I wonder if there is a test we could run or some kind of tool to help us check that at each PR. Sorry for the inconvenience.

It should be fairly easy to fix by checking what differs from other docstrings.

**Comment by dstine** [comment]
This appears to be a widespread issue, a number of methods are affected. At least, those whose Argument section is the last section of the docstring.

It can be "fixed" by adding a newline at the end of the docstring. For example, in `optimizers.py` change from
```
class SGD(Optimizer):
    """Stochastic gradient descent optimizer.

    Includes support for momentum,
    learning rate decay, and Nesterov momentum.

    # Arguments
        lr: float >= 0. Learning rate.
        momentum: float >= 0. Parameter that accelerates SGD
            in the relevant direction and dampens oscillations.
        decay: float >= 0. Learning rate decay over each update.
        nesterov: boolean. Whether to apply Nesterov momentum.
    """
```
to
```
class SGD(Optimizer):
    """Stochastic gradient descent optimizer.

    Includes support for momentum,
    learning rate decay, and Nesterov momentum.

    # Arguments
        lr: float >= 0. Learning rate.
        momentum: float >= 0. Parameter that accelerates SGD
            in the relevant direction and dampens oscillations.
        decay: float >= 0. Learning rate decay over each update.
        nesterov: boolean. Whether to apply Nesterov momentum.

    """
```

But that's probably not the way to proceed. It would require changing a number of existing docstrings.

As far as I can tell by generating the docs with various code versions, the root cause is a recent change to `autogen.py` itself:
1) I observe the problem with dcb5e3cf - master from today.
2) I observe the problem with a37b41d5 - master from October 29.
3) I do not observe the problem with aa0f12ce - the parent commit of a37b41d5.

Certainly I could be wrong, but it seems that #11508 caused this regression.

I looked at the code a bit, but could not immediately determine how to change it.

**Comment by dstine** [comment]
Related general docs questions: Under what circumstances are the live keras.io docs published? Probably with each new release? Am I missing an indicator of the Keras version corresponding to the live docs?

**Comment by dstine** [comment]
fc327b952f09c9229ee6a6bd1ed1b47d2e0dc3f5 over in my fork has a test that
1) fails for dcb5e3cf
2) fails for a37b41d5
3) passes for aa0f12ce

**Comment by gabrieldemarmiesse** [comment]
Thank you for the investigation. This is very helpful. Yes, the docs are published after each new release.

I think our best option would be more tests, and a better algorithm to extract the docstring and put it in markdown. Currently, the algorithm to do this is quite complicated. I believe it could be simplified. Would you want to make a PR?

**Comment by gabrieldemarmiesse** [comment]
The root of the problem is more the function `process_docstring` in `autogen.py`. It's very complex, error prone, we can't easily test it. In the end, I believe there must be a better solution.
